overlord/ifacestate/ifacestate.go

```diff
@@ -160,13 +173,16 @@ func connect(st *state.State, change *state.Change, mainTask *state.Task, plugSn
 
 	connectInterface.Set("slot", interfaces.SlotRef{Snap: slotSnap, Name: slotName})
 	connectInterface.Set("plug", interfaces.PlugRef{Snap: plugSnap, Name: plugName})
-	connectInterface.Set("auto", mainTask != nil && mainTask.Kind() == "auto-connect")
+	connectInterface.Set("auto", installedSnapTask != nil && installedSnapTask.Kind() == "auto-connect")
 
 	// Expose a copy of all plug and slot attributes coming from yaml to interface hooks. The hooks will be able
 	// to modify them but all attributes will be checked against assertions after the hooks are run.
-	if err := setInitialConnectAttributes(connectInterface, plugSnap, plugName, slotSnap, slotName); err != nil {
-		return nil, err
-	}
+	emptyDynamicAttrs := map[string]interface{}{}
+	connectInterface.Set("plug-static", plugStatic)
+	connectInterface.Set("slot-static", slotStatic)
+	connectInterface.Set("plug-dynamic", emptyDynamicAttrs)
```

Comment:
as we discussed with this is a bit unclear how the snaps can distinguish initial connect vs reconnect and reuse/avoid leaking resources  without rolling their own handshake and bookkeeping state